functions.php

```diff
@@ -66,6 +67,23 @@ function wpm_is_type( $type, $value ) {
 		}
 	}
 
+	// Check if the type is an array of a certain type.
+	if ( '[]' === substr( $type, -2 ) ) {
```

Comment:
@CrochetFeve0251 The union type is managed at the beginning of `wpm_apply_filters_typed` so that, when entering `wpm_is_type`, `$type` is a single type I think (for instance `string` or `string[]` or `false`)